src/main/java/favourite/use_case/PostFactory.java

```diff
@@ -0,0 +1,63 @@
+package favourite.use_case;
+
+import entities.Post;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * PostFactory in the use case layer implements PostReaderInterface.
+ * This class turns data from the database into Post entities the FavouriteInteractor can directly interact with.
+ */
+public class PostFactory implements PostReaderInterface{
+
+    /**
+     * Creates a new Post object based on the data stored in the database
+     * @param postdata Array of Strings, data stored in the database for a post
+     * @return Post object with qualities described in postdata
+     */
+    @Override
+    public Post readPost(String[] postdata) {
+        // converting string data into acceptable types to reconstruct a Post object
+        int posterID = Integer.parseInt(postdata[1]);
+        String title = postdata[2];
+        String mainDesc = postdata[3];
+        String[] tagarray = postdata[4].split(" ");
+        List<String> tags = new ArrayList<>();
+        for (String tag : tagarray) {
+            tags.add(tag);
+        }
+        String collaborators = postdata[5];
+        // creating LocalDate object for deadline
+        String[] deadlineElements = postdata[6].split(" ");
+        int year1 = Integer.parseInt(deadlineElements[0]);
```

Comment:
To make your life easier here, you can use LocalDate.parse(String date) instead! The date has to be in the format yyyy-MM-dd though, which should be the case in the database! It will automatically throw an error if the date does not exist, e.g. February 31st

Reply:
this is great, thank you